.github/workflows/image.yaml

```diff
@@ -0,0 +1,46 @@
+
+name: Build the operator image and push to public ECR on the main branch
+on:
+  push:
+    branches: [main]
+jobs:
+  build-push-docker-image:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      - uses: actions/cache@v2
+        with:
+          path: |
+            ~/.cache/go-build
+            ~/go/pkg/mod
+            ~/go/bin/
+          key: ${{ runner.os }}-go-${{ hashFiles('**/go.sum') }}
+          restore-keys: |
+            ${{ runner.os }}-go-
+      - uses: actions/setup-go@v2
+        with:
+          go-version: 1.16
+      - name: Build & Test
+        run: |
+          go build -v ./cmd/controller
+          go test ./...
+        working-directory: operator
+      - uses: docker/login-action@v1
+        with:
+          registry: public.ecr.aws
+          username: ${{ secrets.AWS_ACCESS_KEY_ID }}
+          password: ${{ secrets.AWS_SECRET_ACCESS_KEY }}
+        env:
+          AWS_REGION: us-east-1
+      - name: Install KO binary
```

Comment:
You should be able to use `go install github.com/google/ko` like Karpenter's toolchain script.

Reply:
I tried it, it takes time(40-50s) to build the and install. Getting the binary is much quicker.